distributed/tests/test_nanny.py

```diff
@@ -290,6 +290,58 @@ def leak():
         assert "memory" in out.lower()
 
 
+@gen_cluster(
+    nthreads=[("127.0.0.1", 1)] * 8,
+    client=True,
+    Worker=Nanny,
+    worker_kwargs={"memory_limit": 2e8, "memory_pause_fraction": 0.6},
+    timeout=20,
+    clean_kwargs={"threads": False},
+)
+def test_nanny_throttle(c, s, *workers):
```

Comment:
You might enjoy using `async def` here, and replacing `yield`s with `await`s.  It's not important, but I personally find it satisfying for some reason :)